diff --git a/django/core/files/uploadhandler.py b/django/core/files/uploadhandler.py
--- a/django/core/files/uploadhandler.py
+++ b/django/core/files/uploadhandler.py
@@ -113,7 +113,7 @@
         Signal that a file has completed. File size corresponds to the actual
         size accumulated by all the chunks.
 
-        Subclasses must should return a valid ``UploadedFile`` object.
+        Subclasses should return a valid ``UploadedFile`` object.
         """
         raise NotImplementedError()
 
