diff -r d582d5fdc256 -r f61dbbfd2ba5 src/dwb.c
--- a/src/dwb.c Tue Nov 02 23:43:04 2010 +0100
+++ b/src/dwb.c Wed Nov 03 16:11:04 2010 +0100
@@ -756,9 +756,8 @@
 /* dwb_test_cookie_allowed(const gchar *)     return:  gboolean{{{*/
 static gboolean
 dwb_test_cookie_allowed(SoupCookie *cookie) {
-  const gchar *domain = cookie->domain;
   for (GList *l = dwb.fc.cookies_allow; l; l=l->next) {
-    if (!strcmp(domain, l->data)) {
+    if(soup_cookie_domain_matches(cookie, l->data)) {
       return true;
     }
   } 
